savoirfairelinux/sous-chef
# Fix #599

## Fixes #599  by lingxiaoyang

### Changes proposed in this pull request:

* Define model property `Order_item.is_a_client_bill` if it is labeled "delivery" with no meal components. (Note: in earlier version of code, in order batch creation, we have "delivery" as the default type of every item. It has been corrected to "meal_component")
* Define model property and setter `Order.includes_a_bill`.
* Modify frontend: add billing toggle in route sheet table (in print, just yes/no). Add JS interaction for this toggle and its API view in Django.
* Add unit tests for all three above.

### Status

- [ ] HOLD wait PR~~~s #592 and~~~ #593 to be merged

### How to verify this change

http://localhost:8000/delivery/route_sheet/1/

The "Bill" toggles should be visible. Go to order detail view, toggle "Bill", refresh order detail view, then we should observe the addition/deletion of "delivery" item ($0.00, non-billable, no meal_component).

http://localhost:8000/delivery/routes/?print=yes

The print view should show Bill status (yes/no).

### Deployment notes and migration

none

### New translatable strings


### Additional notes


![image](https://cloud.githubusercontent.com/assets/8630726/21402008/945f1e00-c784-11e6-8ddb-a83ff6c72291.png)

![image](https://cloud.githubusercontent.com/assets/8630726/21402017/a1ffd86a-c784-11e6-97ca-78fa175477df.png)

![image](https://cloud.githubusercontent.com/assets/8630726/21402029/b02742b6-c784-11e6-8119-a31c7de5ec34.png)
